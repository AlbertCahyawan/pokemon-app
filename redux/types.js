export const SET_POKE_LIST = "SET_POKE_LIST";
