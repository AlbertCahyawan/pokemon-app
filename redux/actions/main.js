import * as t from "../types";

export const setPokeList = (data) => dispatch => {
  dispatch({
    type: t.SET_POKE_LIST,
    payload: data
  });
}
