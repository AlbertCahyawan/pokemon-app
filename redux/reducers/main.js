import * as t from "../types";

const main = (state = {
    data: {},
    offset: 0,
    limit: 12,
    firstLoad:false,
}, action) => {
  switch(action.type){
    case t.SET_POKE_LIST:
      return { 
        ...state,
        data: action.payload,
        offset: state.offset + state.limit ,
        firstLoad: true
      };
    default:
      return {...state};
    }
}

export default main;