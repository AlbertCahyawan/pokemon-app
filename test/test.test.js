// import React from 'react';
// import renderer from 'react-test-renderer';
// import PokemonCard from '../components/pokemonCard';

// test('Link changes the class when hovered', () => {
//   const component = renderer.create(
//     <Link page="http://www.facebook.com">Facebook</Link>,
//   );
//   let tree = component.toJSON();
//   expect(tree).toMatchSnapshot();

//   // manually trigger the callback
//   tree.props.onMouseEnter();
//   // re-rendering
//   tree = component.toJSON();
//   expect(tree).toMatchSnapshot();

//   // manually trigger the callback
//   tree.props.onMouseLeave();
//   // re-rendering
//   tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });

import * as actions from '../redux/actions/main'
import * as t from '../redux/types' 

describe('actions', () => {
  it('should create an action to add a todo', () => {
    const payload = 'test'
    const expectedAction = {
      type: 'SET_POKE_LIST',
      payload
    }
    expect(actions.setPokeList(payload)).toEqual(expectedAction)
  })
})

// import * as actions from './addTodo' 

// describe('actions', () => {
//   it('should create an action to add a todo', () => {
//     const text = 'Finish docs'
//     const expectedAction = {
//       type: 'ADD_TODO',
//       text
//     }
//     expect(actions.addTodo(text)).toEqual(expectedAction)
//   })
// })