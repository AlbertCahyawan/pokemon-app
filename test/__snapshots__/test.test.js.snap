// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check snapshot on default 1`] = `
<div
  className="pokemon-card-container"
>
  <div
    className="pokemon-display"
  >
    <img
      alt="Bulbasaur"
      src="https://assets.pokemon.com/assets/cms2/img/pokedex/detail/001.png"
    />
  </div>
  <div
    className="pokemon-card-description"
  >
    <div
      className="description-id"
    >
      #
      1
    </div>
    <h2
      className="description-Name"
    >
      Bulbasaur
    </h2>
  </div>
</div>
`;

exports[`check snapshot on default 2`] = `
<div
  className="pokemon-card-container"
>
  <div
    className="pokemon-display"
  >
    <img
      alt="Bulbasaur"
      src="https://assets.pokemon.com/assets/cms2/img/pokedex/detail/001.png"
    />
  </div>
  <div
    className="pokemon-card-description"
  >
    <div
      className="description-id"
    >
      #
      1
    </div>
    <h2
      className="description-Name"
    >
      Bulbasaur
    </h2>
  </div>
</div>
`;

exports[`snapshot when there is new changes 1`] = `
<div
  className="pokemon-card-container"
>
  <div
    className="pokemon-display"
  >
    <img
      alt="bulbasaur"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
    />
  </div>
  <div
    className="pokemon-card-description"
  >
    <div
      className="description-id"
    >
      #
      1
    </div>
    <h2
      className="description-Name"
    >
      bulbasaur
    </h2>
    <div
      className="description-button"
    >
      <button
        onClick={[Function]}
      >
        Release
      </button>
    </div>
  </div>
</div>
`;

exports[`snapshot when there is new changes 2`] = `
<div
  className="pokemon-card-container"
>
  <div
    className="pokemon-display"
  >
    <img
      alt="bulbasaur"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
    />
  </div>
  <div
    className="pokemon-card-description"
  >
    <div
      className="description-id"
    >
      #
      1
    </div>
    <h2
      className="description-Name"
    >
      bulbasaur
    </h2>
    <div
      className="description-button"
    >
      <button
        onClick={[Function]}
      >
        Release
      </button>
    </div>
  </div>
</div>
`;
